src/commons/sagas/WorkspaceSaga/index.ts

```diff
@@ -514,15 +469,12 @@ export default function* WorkspaceSaga(): SagaIterator {
           })
         );
       }
-    }
-  );
-
-  yield takeLeading(
-    EVAL_EDITOR_AND_TESTCASES,
-    function* (action: ReturnType<typeof actions.runAllTestcases>) {
+    },
+    // TODO: Should be takeLeading, not takeEvery
```

Comment:
Was this behaviour checked?

Reply:
Will have to test with an actual assessment with testcases. But should be no different to `takeEvery` except for the fact that `takeLeading` is blocking.